packages/labs/compiler/src/lib/template-transform.ts

```diff
@@ -314,13 +314,17 @@ class CompiledTemplatePass {
           }
           if (node.attrs.length > 0) {
             for (const attr of node.attrs) {
-              if (
-                attr.name.endsWith(boundAttributeSuffix) ||
-                attr.name.startsWith(marker)
-              ) {
+              const isAttributePart = attr.name.endsWith(boundAttributeSuffix);
+              const isElementPart = attr.name.startsWith(marker);
+              if (isAttributePart || isElementPart) {
                 attributesToRemove.add(attr);
-                const realName = attrNames[attrNameIndex++];
-                if (realName !== undefined) {
+                if (isAttributePart) {
+                  const realName = attrNames[attrNameIndex++];
+                  if (!realName) {
```

Comment:
Nit: on behalf of Justin
```suggestion
                  if (realName === undefined) {
```

Reply:
Appreciated!